src/Hadrian/Utilities.hs

```diff
@@ -481,3 +484,6 @@ renderUnicorn ls =
     ponyPadding = "                                            "
     boxLines :: [String]
     boxLines = ["", "", ""] ++ (lines . renderBox $ ls)
+
+data TestSpeed = Slow | Average | Fast deriving (Show, Eq)
```

Comment:
I'm not sure `TestSpeed` is a generally useful data type. It's certainly useful for the GHC build system, but other build systems may have different settings or may not even need a testsuite (remember, `Hadrian.*` namespace is for generic functionality that may be useful not only for building GHC).

I suggest to move `TestSpeed` to the `CommandLine` module.